### PR TITLE
ForbitTStruct handles nilable properties

### PR DESCRIPTION
**Pull request by Morriar**
Closes https://github.com/Shopify/rubocop-sorbet/issues/179.

### CODE REVIEW

File: spec/rubocop/cop/sorbet/forbid_t_struct_spec.rb [hunk 1]
```diff
@@ -374,5 +374,39 @@ def initialize(foo:, bar:)
...
+          sig { returns(T.nilable(Integer)) }
+          attr_reader :foo
+
+          sig { returns(T.nilable(String)) }
+          attr_accessor :bar
+
+          sig { returns(Integer) }
+          attr_accessor :baz
+
+          sig { params(baz: Integer, foo: T.nilable(Integer), bar: T.nilable(String)).void }
```
[the remark this answers is not in the document]
Comment (KaanOzkan) [reply spec/rubocop/cop/sorbet/forbid_t_struct_spec.rb:400]: Order is kept for keyword arguments `foo` and `bar`.  Only `baz` is moved earlier.

**Review by KaanOzkan — APPROVED** [review]

File: spec/rubocop/cop/sorbet/forbid_t_struct_spec.rb [hunk 2]
```diff
@@ -374,5 +374,39 @@ def initialize(foo:, bar:)
 
       expect(autocorrect_source(source)).to(eq(corrected))
     end
+
+    it "handles optional nilable properties" do
+      source = <<~RUBY
+        class Foo < T::Struct
+          const :foo, T.nilable(Integer)
```
[the remark this answers is not in the document]
Comment (Morriar) [reply spec/rubocop/cop/sorbet/forbid_t_struct_spec.rb:381]: Yeah, we do best effort here 👍

File: spec/rubocop/cop/sorbet/forbid_t_struct_spec.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (Morriar) [reply spec/rubocop/cop/sorbet/forbid_t_struct_spec.rb:400]: This is a rule within Sorbet, required keyword parameters must be before the optional ones: https://sorbet.run/#%23%20typed%3A%20true%0Aextend%20T%3A%3ASig%0A%0Asig%20%7Bparams%28x%3A%20Integer%2C%20y%3A%20Integer%29.void%7D%0Adef%20foo%28x%3A%2042%2C%20y%3A%29%0A%20%20puts%28x%20%2B%201%29%0Aend

**Review by paracycle — APPROVED** [review]

**Comment by andyw8** [comment]
I fixed the typo in the PR title in case anyone is searching for this.

**Pull request merged**